src/ocean_emulators/datasets.py

```diff
@@ -604,6 +416,7 @@ def __init__(
         executor: ThreadPoolExecutor | None = None,
     ):
         super().__init__()
+        self.id = self.__class__.__name__ + "_" + str(id(self))
```

Comment:
🐑 nit: this is slightly faster/more canonical:
```suggestion
        self.id = f"{self.__class__.__name__}_{str(id(self))}"
```
(I don't know if `str` should be there or not)